src/main/kotlin/net/maizegenetics/phgv2/pathing/BuildKmerIndex.kt

```diff
@@ -123,23 +138,22 @@ class BuildKmerIndex: CliktCommand(help="Create a kmer index for a HaplotypeGrap
      * This returns a HashMap of hash -> hapid list for all the kmers in the keep set.
      * Which allows the export to not need to do a second pass over the sequences to get the set of hapIds which contain the unique kmers.
      */
-    fun processGraphKmers(graph: HaplotypeGraph, dbPath: String, maxHaplotypeProportion: Double=.75,
-                                        hashMask: Long = 3, hashFilterValue:Long = 1) : Long2ObjectOpenHashMap<Set<String>> {
+    fun processGraphKmersNew(graph: HaplotypeGraph, dbPath: String, maxHaplotypeProportion: Double=.75,
+                          hashMask: Long = 3, hashFilterValue:Long = 1) : Pair<Long2ObjectOpenHashMap<Set<String>>,LongOpenHashBigSet> {
         //keepMap is a map of hash -> Set of haplotype ids
-        val keepMap = Long2ObjectOpenHashMap<Set<String>>()
+        //TODO Change this to be more RAM efficient.  Probably need something like this:
+        // Map<Long, Set<Pair<refRangeIdx, hapIdBitset>>
+        val keepMap = Long2ObjectOpenHashMap<MutableSet<String>>(500_000_000)
 
         //discardSet is a Set of hashes
-        val discardSet = LongOpenHashSet()
+        val discardSet = LongOpenHashBigSet(500_000_000)
         val startTime = System.nanoTime()
         val sampleGametes = graph.sampleGametesInGraph()
 
         val contigRangesMap = graph.rangesByContig()
 
-        //needed for diagnostics
-        val refRangeToIndexMap = graph.refRangeToIndexMap()
-        val hapidToRefrangeMap = graph.hapIdToRefRangeMap()
-
-        val maxHapsToKeep = sampleGametes.size * 2
+        //This is a parameter we are currently testing
```

Comment:
Maybe a slightly expanded comment as to why we only want to keep sampleGametee.size * .5 vs the previous *2.  A general comment on why a smaller proportion of them is better than a greater proportion